Tests/MongoSwiftTests/ReadPreferenceTests.swift

```diff
@@ -0,0 +1,89 @@
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+final class ReadPreferenceTests: XCTestCase {
+    static var allTests: [(String, (ReadPreferenceTests) -> () throws -> Void)] {
+        return [
+            ("testReadPreferenceMode", testReadPreferenceMode),
+            ("testReadPreferenceTagSets", testReadPreferenceTagSets),
+            ("testReadPreferenceTagSets", testReadPreferenceMaxStalenessSeconds)
+        ]
+    }
+
+    override func setUp() {
+        self.continueAfterFailure = false
+    }
+
+    func testReadPreferenceMode() {
+        let primary = ReadPreference(.primary)
+        expect(primary.mode).to(equal(.primary))
```

Comment:
Travis [does not like this](https://travis-ci.org/mongodb/mongo-swift-driver/builds/399310940#L1932):

```
/home/travis/build/mongodb/mongo-swift-driver/Tests/MongoSwiftTests/ReadPreferenceTests.swift:20:33: error: ambiguous reference to member 'equal'
        expect(primary.mode).to(equal(.primary))
                                ^~~~~
```

Swift 4.1 didn't seem to have a problem with this, but I assume that's due to some subtle improvement since 4.0. Will change so that `equal()` refers to the full enum (e.g. `ReadPreference.Mode.primary`).